File: 10.1D/main/java/com/example/personalizedlearningexperiences/signup.java
package com.example.personalizedlearningexperiences;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.example.personalizedlearningexperiences.databinding.FragmentSignupBinding;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link signup#newInstance} factory method to
 * create an instance of this fragment.
 */
public class signup extends Fragment {

    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    // TODO: Rename and change types of parameters
    private String mParam1;
    private String mParam2;
    private FragmentSignupBinding binding;

    public signup() {
        // Required empty public constructor
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param param1 Parameter 1.
     * @param param2 Parameter 2.
     * @return A new instance of fragment signup.
     */
    // TODO: Rename and change types and number of parameters
    public static signup newInstance(String param1, String param2) {
        signup fragment = new signup();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment

        binding = FragmentSignupBinding.inflate(inflater,container, false);

        binding.interestButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.i("Ash","Switch");
                if (binding.editUsername.getText().toString().isEmpty()) {
                    // Username is empty, show toast message
                    Toast.makeText(getActivity().getApplicationContext(), "Username is required", Toast.LENGTH_SHORT).show();
                    return;
                }

                if (binding.editEmail.getText().toString().isEmpty()) {
                    // Email is empty, show toast message
                    Toast.makeText(getActivity().getApplicationContext(), "Email is required", Toast.LENGTH_SHORT).show();
                    return;
                }

                if (binding.editConfirmEmail.getText().toString().isEmpty()) {
                    // Confirm Email is empty, show toast message
                    Toast.makeText(getActivity().getApplicationContext(), "Confirm Email is required", Toast.LENGTH_SHORT).show();
                    return;
                }

                if (binding.editPassword.getText().toString().isEmpty()) {
                    // Password is empty, show toast message
                    Toast.makeText(getActivity().getApplicationContext(), "Password is required", Toast.LENGTH_SHORT).show();
                    return;
                }

                if (binding.editConfirmPassword.getText().toString().isEmpty()) {
                    // Confirm Password is empty, show toast message
                    Toast.makeText(getActivity().getApplicationContext(), "Confirm Password is required", Toast.LENGTH_SHORT).show();
                    return;
                }

                if (binding.editPhone.getText().toString().isEmpty()) {
                    // Phone is empty, show toast message
                    Toast.makeText(getActivity().getApplicationContext(), "Phone Number is required", Toast.LENGTH_SHORT).show();
                    return;
                }

                if (!binding.editEmail.getText().toString().equals(binding.editConfirmEmail.getText().toString())) {
                    Toast.makeText(getActivity().getApplicationContext(), "Emails must match", Toast.LENGTH_SHORT).show();
                    return;
                }

                if (!binding.editPassword.getText().toString().equals(binding.editConfirmPassword.getText().toString())) {
                    Toast.makeText(getActivity().getApplicationContext(), "Passwords must match", Toast.LENGTH_SHORT).show();
                    return;
                }

                // All fields are filled, proceed with other actions

                // Create a bundle to pass data
                Bundle bundle = new Bundle();
                bundle.putString("username", binding.editUsername.getText().toString());
                bundle.putString("email", binding.editEmail.getText().toString());
                bundle.putString("confirmEmail", binding.editConfirmEmail.getText().toString());
                bundle.putString("password", binding.editPassword.getText().toString());
                bundle.putString("confirmPassword", binding.editConfirmPassword.getText().toString());
                bundle.putString("phone", binding.editPhone.getText().toString());

                // Navigate to the next fragment with the bundle
                Navigation.findNavController(v).navigate(R.id.action_signup_to_interests, bundle);

            }
        });

        return binding.getRoot();
    }
}

File: 5.1C iTube/java/com/example/itube/signup.kt
package com.example.itube

import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.lifecycle.lifecycleScope
import androidx.navigation.fragment.findNavController
import com.example.itube.databinding.FragmentSignupBinding
import kotlinx.coroutines.launch

// TODO: Rename parameter arguments, choose names that match
// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
private const val ARG_PARAM1 = "param1"
private const val ARG_PARAM2 = "param2"

/**
 * A simple [Fragment] subclass.
 * Use the [signup.newInstance] factory method to
 * create an instance of this fragment.
 */
class signup : Fragment() {
    // TODO: Rename and change types of parameters
    private var param1: String? = null
    private var param2: String? = null
    private lateinit var binding:FragmentSignupBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            param1 = it.getString(ARG_PARAM1)
            param2 = it.getString(ARG_PARAM2)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment

        binding = FragmentSignupBinding.inflate(inflater, container, false)
        val usernames = mutableListOf<String>()
        for(user in users){
            usernames.add(user.username.toString())
        }


        binding.createAccountButton.setOnClickListener{
            if (binding.editUserSignup.text.toString() !in usernames){
                if (binding.editPassSignup.text.toString() == binding.editConfirmPass.text.toString()){
                    if (binding.editPassSignup.text.toString() != "" && binding.editUserSignup.text.toString() != "" && binding.editName.text.toString() != ""){
                        val newPlaylist: MutableList<String> = ArrayList()
                        val newUser = User(binding.editUserSignup.text.toString(),binding.editPassSignup.text.toString(),newPlaylist)
                        lifecycleScope.launch {
                            database.userDao().insertUser(newUser)
                            findNavController().navigate(R.id.action_signup_to_login)
                        }
                    }
                    else{
                        Toast.makeText(this.context,"All fields must be filled",Toast.LENGTH_SHORT).show()
                    }
                }
                else{
                    Toast.makeText(this.context,"Passwords don't match",Toast.LENGTH_SHORT).show()
                }
            }
            else{
                Toast.makeText(this.context,"Username already exits",Toast.LENGTH_SHORT).show()
            }
        }

        return binding.root
    }

    companion object {
        /**
         * Use this factory method to create a new instance of
         * this fragment using the provided parameters.
         *
         * @param param1 Parameter 1.
         * @param param2 Parameter 2.
         * @return A new instance of fragment signup.
         */
        // TODO: Rename and change types and number of parameters
        @JvmStatic
        fun newInstance(param1: String, param2: String) =
            signup().apply {
                arguments = Bundle().apply {
                    putString(ARG_PARAM1, param1)
                    putString(ARG_PARAM2, param2)
                }
            }
    }
}